test/libsolidity/StandardCompiler.cpp

```diff
@@ -1782,6 +1924,346 @@ BOOST_AUTO_TEST_CASE(source_location_of_bare_block)
 	BOOST_REQUIRE(sourceMap.find(sourceRef) != std::string::npos);
 }
 
+BOOST_AUTO_TEST_CASE(ethdebug_excluded_from_wildcards)
+{
+	frontend::StandardCompiler compiler;
+	// excluded from output selection wildcard
+	Json result = compiler.compile(generateStandardJson(true, {}, Json::array({"*"})));
+	BOOST_REQUIRE(result.dump().find("ethdebug") == std::string::npos);
+	// excluded from debug info selection wildcard
+	result = compiler.compile(generateStandardJson(true, {"*"}, Json::array({"ir"})));
+	BOOST_REQUIRE(result.dump().find("ethdebug") == std::string::npos);
+	// excluded from both - just in case ;)
+	result = compiler.compile(generateStandardJson(true, {"*"}, Json::array({"*"})));
+	BOOST_REQUIRE(result.dump().find("ethdebug") == std::string::npos);
+}
+
+BOOST_AUTO_TEST_CASE(ethdebug_debug_info_ethdebug)
+{
+	static std::vector<std::tuple<Json, std::string, std::optional<std::function<bool(Json)>>>> tests{
+		{
+			generateStandardJson(false, Json::array({"ethdebug"}), Json::array({"*"})),
+			"'settings.debug.debugInfo' can only include 'ethdebug', if output 'ir', 'irOptimized', 'evm.bytecode.ethdebug' or 'evm.deployedBytecode.ethdebug' was selected.",
+			std::nullopt,
+		},
+		{
+			generateStandardJson(true, Json::array({"ethdebug"}), Json::array({"*"})),
+			"'settings.debug.debugInfo' can only include 'ethdebug', if output 'ir', 'irOptimized', 'evm.bytecode.ethdebug' or 'evm.deployedBytecode.ethdebug' was selected.",
+			std::nullopt,
+		},
+		{
+			generateStandardJson(false, Json::array({"ethdebug"}), Json::array({"evm.bytecode.ethdebug"})),
+			"'evm.bytecode.ethdebug' or 'evm.deployedBytecode.ethdebug' can only be selected as output, if 'viaIR' was set.",
+			std::nullopt,
+		},
+		{
+			generateStandardJson(false, Json::array({"ethdebug"}), Json::array({"evm.deployedBytecode.ethdebug"})),
+			"'evm.bytecode.ethdebug' or 'evm.deployedBytecode.ethdebug' can only be selected as output, if 'viaIR' was set.",
+			std::nullopt,
+		},
+		{
+			generateStandardJson(false, Json::array({"ethdebug"}), Json::array({"evm.bytecode.ethdebug", "evm.deployedBytecode.ethdebug"})),
+			"'evm.bytecode.ethdebug' or 'evm.deployedBytecode.ethdebug' can only be selected as output, if 'viaIR' was set.",
+			std::nullopt,
+		},
+		{
+			generateStandardJson(false, Json::array({"ethdebug"}), Json::array({"ir"})),
```

Comment:
ah true! thanks for the reminder, I wanted to at some basic smoke tests

Reply:
It was definitely worth to add those tests, because it revealed a small bug in cli parameter handling in the context of strict assembly :) also with yul in standard json the top-level ethdebug object was missing in the output